src/placeos-models/utilities/encryption.cr

```diff
@@ -27,30 +27,36 @@ module PlaceOS::Encryption
   private SECRET = ENV["PLACE_SERVER_SECRET"]?.tap { |k| Log.warn { "using insecure default secret" } if k.nil? } || "super secret, do not leak"
   private CIPHER = "aes-256-gcm"
 
+  # Encrypt a test string with the encryption context in the cipher context
+  # and test for equality
+  #
+  def self.check?(encrypted : String, test : String, id : String, level : Level) : Bool
+    return true if level == Level::None && encrypted == test
+    return false unless is_encrypted?(encrypted)
+
+    salt, iv, cipher_text = extract_context(encrypted)
+
+    _, key = generate_key(id: id, level: level, salt: salt)
+
+    encrypted_test = encrypt_data(data: test, key: key, salt: salt, iv: ::Base64.decode(iv))
+    encrypted_test[:cipher_text] == cipher_text
```

Comment:
```suggestion
    Crypto::Subtle.constant_time_compare encrypted_test[:cipher_text], cipher_text
```
See https://crystal-lang.org/api/0.35.1/Crypto/Subtle.html#constant_time_compare(x,y):Bool-class-method.

Reply:
For now we'll just allow admins to view the secrets.

In the future, we could just shoot the secret back with the create and warn the user they'll need to note the secret for deployment of their edge nodes.